R/Attributes.R

```diff
@@ -543,6 +543,23 @@ compileAttributes <- function(pkgdir = ".", verbose = getOption("verbose")) {
         list(env = list(PKG_CXXFLAGS ="-std=c++17"))
 }
 
+# built-in C++20 plugin for C++20
+.plugins[["cpp20"]] <- function() {
+    if (getRversion() >= "4.2")         # with recent R versions, R can decide
+        list(env = list(USE_CXX20 = "yes"))
+    else
+        list(env = list(PKG_CXXFLAGS ="-std=c++20"))
```

Comment:
I think it should be `PKG_CXXFLAGS ="-std=c++2a"` for experimental support.

Reply:
Hm.  Note that this is an `else` branch and by default we don't even get there.  The higher-level abstraction of letting R take care of how it calls it via `USE_CXX20` is good.   

And both tests (for 20 and 23) worked with the `g++-11` I have here which is not even all that new -- I didn't yet dive into a quick Debian container to try `g++-12`.

Look winded way of saying:   do we really need it?   If someone (you?) wants to set `PKG_CXXFLAGS ="-std=c++2a"` they can outside the plugin.  Unless I miss something I think the default I put here is good.

Reply:
The thing is that there are a couple of machines with gcc 8 on CRAN, which doesn't support the `-std=c++20` flag, but the experimental `-std=c++2a`. So a package using this plugin will fail in those oldrel machines.

Reply:
Well, to be blunt, strictly not my problem if people opt into a standard their compiler does not support. 

And I do not think because of that we should only enable 'experimental' sub-support.  People use C++17 in production, are moving to C++20 and will before long use C++23 (with g++ 11, 12, ...).  We should support that.

If you look further down in the file we also used to add plugins for these half-baked but not quite standards.  We could do that but I still think such users can still fiddle PKG_CXXFLAGS as they see fit.

Reply:
I was simply talking about conforming to current CRAN support because that's what was done in the past with C++11 (as can be seen in the several branches for the `cpp11` plugin). But I'm ok if we don't want to do that anymore.

Reply:
I think in the past the compiler barked and only offered the partial/experimental support via the 'a' or 'b' variant.  That's why I tried to test this, and unless I missed something the test passed.  So ... good enough?